src/unix/utilsx11.cpp

```diff
@@ -2607,27 +2607,35 @@ static bool wxGetKeyStateGTK(wxKeyCode key)
 
     GdkDisplay* display = gdk_window_get_display(wxGetTopLevelGDK());
     GdkKeymap* keymap = gdk_keymap_get_for_display(display);
-    guint state = gdk_keymap_get_modifier_state(keymap);
-    guint mask = 0;
-    switch (key)
-    {
-        case WXK_ALT:
-            mask = GDK_MOD1_MASK;
-            break;
 
-        case WXK_CONTROL:
-            mask = GDK_CONTROL_MASK;
-            break;
+    if (key == WXK_ALT)
+        return (gdk_keymap_get_modifier_state(keymap) & GDK_MOD1_MASK) != 0;
 
-        case WXK_SHIFT:
-            mask = GDK_SHIFT_MASK;
-            break;
+    if (key == WXK_CONTROL)
+        return (gdk_keymap_get_modifier_state(keymap) & GDK_CONTROL_MASK) != 0;
 
-        default:
-            wxFAIL_MSG(wxS("Unsupported key, only modifiers can be used"));
-            return false;
-    }
-    return (state & mask) != 0;
+    if (key == WXK_SHIFT)
+        return (gdk_keymap_get_modifier_state(keymap) & GDK_SHIFT_MASK) != 0;
+
+    if (key == WXK_CAPITAL)
+        return gdk_keymap_get_caps_lock_state(keymap) != FALSE;
+
+    if (key == WXK_NUMLOCK)
+        return gdk_keymap_get_num_lock_state(keymap) != FALSE;
+
+# if GTK_CHECK_VERSION(3,18,0)
```

Comment:
We need both compile- and run-time version checks using `gtk_check_version()`, as is done elsewhere, otherwise the program compiled on a current system but running on the old one would just crash.

Reply:
But what runtime check is missing? There is (was) if (gtk_check_version(3,4,0) != nullptr) in very beginning of wxGetKeyStateGTK that automatically ensures all functions inside except gdk_keymap_get_scroll_lock_state. And gdk_keymap_get_scroll_lock_state has corresponding gtk_check_version(3,18,0).

Reply:
Regarding using switch - i removed it to avoid using of gdk_keymap_get_modifier_state where its not needed. Well, it was possible to add another switch for functions that dont need gdk_keymap_get_modifier_state but for me two switch-es would look more tricky than sequence of if-s...

Reply:
Sorry, I'm blind. Or so used to seeing `gtk_check_version()` check on its own line that I somehow skipped it below. No problem here, sorry again.